src/ORM/Table.php

```diff
@@ -700,14 +700,33 @@ public function setDisplayField($field)
     public function getDisplayField()
     {
         if ($this->_displayField === null) {
+            $displayFieldFound = false;
             $schema = $this->getSchema();
             $this->_displayField = $this->getPrimaryKey();
-            foreach (['title', 'name', 'label'] as $field) {
+            $expectedFields = ['title', 'name', 'label'];
+            foreach ($expectedFields as $field) {
                 if ($schema->hasColumn($field)) {
                     $this->_displayField = $field;
+                    $displayFieldFound = true;
                     break;
                 }
             }
+            if (!$displayFieldFound) {
+                foreach ($expectedFields as $field) {
+                    foreach ($schema->columns() as $column) {
+                        $columnSchema = $schema->getColumn($column);
+                        if (
+                            $columnSchema &&
+                            $columnSchema['null'] !== true &&
+                            $columnSchema['type'] === 'string' &&
+                            strpos($column, $field) !== false
```

Comment:
This is unnecessary, just pick the non null string field irrespective of the name. In most cases people have the column they want as display field as the 1st one.

Reply:
I don't have a good feeling about this but lets see